src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.cs

```diff
@@ -149,5 +151,13 @@ internal void InitializeExtendedAttributesWithExisting(IEnumerable<KeyValuePair<
             TarEntryFormat.Gnu => GnuVersion,
             _ => string.Empty,
         };
+
+        // Stores the archive stream's position where we know the current entry's data section begins,
+        // if the archive stream is seekable. Otherwise, -1.
+        private static void SetDataOffset(TarHeader header, Stream archiveStream) => header._dataOffset =
+            archiveStream.CanSeek
+            // Add one because the last byte read is still part of the header
+            ? archiveStream.Position + 1
```

Comment:
Presumably we could implement this even if the stream isn't seekable by just keeping track of how many bytes we read until this point. Not worthwhile?

Reply:
We could, for sure, but if the underlying stream isn't seekable anyway, would DataOffset be useful?

If this is requested in the future, we could add the functionality, but we would have to hear a good argument or a good use case.